src/Castle.Windsor/MicroKernel/Releasers/LifecycledComponentsReleasePolicy.cs

```diff
@@ -169,12 +194,12 @@ public virtual void Track(object instance, Burden burden)
 						"Release policy was asked to track object '{0}', but its burden has 'RequiresPolicyRelease' set to false. If object is to be tracked the flag must be true. This is likely a bug in the lifetime manager '{1}'.",
 						instance, lifestyle));
 			}
+
 			try
 			{
-				using (@lock.ForWriting())
-				{
-					instance2Burden.Add(instance, burden);
-				}
+				@lock.EnterWriteLock();
```

Comment:
In other places, the code entering the lock is placed outside the `try`. Why not here?